### DISCUSSION TITLE
Most efficient way to get All Amazon Fees

### DISCUSSION THREAD

**Discussion by sdaeric19** [saleweaver/python-amazon-sp-api#48] [Q&A]
Hey guys, 

I'm looking into various endpoints and just wondering if anyone has found the most efficient way to get all AMZ fees (listing fees, FBA fees, storage fees, inbound shipment fees, service fees etc) related to all sales made so far (would be great if it can be split by ASIN too)? 

Do I need to create some reports?

Best wishes,

**Comment by saleweaver** [comment]
Since the fees will vary based on price I would take the finances api and its endpoints. From there you will get every transaction including fees. It's annoying to use the finances api because it's very complicated at a first glance, but once you have all the fields you need to pull it will give you the only real view over fees and everything.

If you want to get estimated fees, reports or the fees endpoints are the way to go.

**Reply by sdaeric19** [reply]
Thank you so much @saleweaver 

So Finances API includes all the fees? 

Also, only one endpoint from the Finances API is covered in this library, right? 

If we need to obtain the order ID and make a call for each order - this could be computationally expensive right?

**Reply by saleweaver** [reply]
The finances api has [4 endpoints](https://github.com/amzn/selling-partner-api-docs/blob/main/references/finances-api/financesV0.md#listfinancialevents), the most interesting one for you will be the last, [listFinancialEvents](https://github.com/amzn/selling-partner-api-docs/blob/main/references/finances-api/financesV0.md#listfinancialevents). Using this you'll get everything in a few calls. 

I've currently only implemented listFinancialEventsByOrderId, feel free to submit a pull request with the new endpoint. I can have a look next week to do it myself, but I'm quite busy atm with projects so it might be a well into next week that I can have a look.
